src/osd/OSD.cc

```diff
@@ -4811,7 +4811,7 @@ PGRef OSD::handle_pg_create_info(const OSDMapRef& osdmap,
     acting_primary,
     info->history,
     info->past_intervals,
-    false,
+    true,
```

Comment:
if this fix is a valid one, we would need to revert f9d1c2bcb599d4cb55ebda59cbe3671c34886217 in master and apply this change to master first.